### PR TITLE
Nd4j.java javadoc 

### PR DESCRIPTION
**Pull request by RobAltena**
Refactoring of the javadoc in Nd4j.java. 

All issues that we stumble upon during the refactor are flagged with TODO comments to be taken down in other pull requests.

### CODE REVIEW

File: nd4j/nd4j-backends/nd4j-api-parent/nd4j-api/src/main/java/org/nd4j/linalg/factory/Nd4j.java [hunk 1]
```diff
@@ -4395,33 +4727,29 @@ public static INDArray create(double[] data, int[] shape, int[] stride, long off
...
+    // This method gets it own javadoc and not a @see because it is used  often.
     /**
-     * Create an ndrray with the specified shape
-     *
-     * @param data  the data to use with tne ndarray
-     * @param shape the shape of the ndarray
-     * @return the created ndarray
+     * Create an array withgiven shape and ordering based on a java double array.
+     * @param data java array used for initialisation. Must have at least the number of elements required.
+     * @@param shape desired shape of new array.
```
Comment (AlexDBlack) [inline nd4j/nd4j-backends/nd4j-api-parent/nd4j-api/src/main/java/org/nd4j/linalg/factory/Nd4j.java]: `@@`

File: nd4j/nd4j-backends/nd4j-api-parent/nd4j-api/src/main/java/org/nd4j/linalg/factory/Nd4j.java [hunk 2]
```diff
@@ -5043,37 +5407,47 @@ public static INDArray createSparseCOO(DataBuffer values, DataBuffer indices, lo
...
-     * @return the created ndarray
+     * @param columns number of columns.
+     * @return the created array
      */
     public static INDArray zeros(int columns) {
         return INSTANCE.zeros(columns);
     }
 
+    /**
+     * Creates a 1D array with the specified data tyoe and number of columns initialized with zero.
```
Comment (AlexDBlack) [inline nd4j/nd4j-backends/nd4j-api-parent/nd4j-api/src/main/java/org/nd4j/linalg/factory/Nd4j.java]: `specified data tyoe`
`number of columns initialized with zero.` -> rephrase, sounds like number of columns will be initialized to zero.
"number of columns, with values initialized to zero"

File: nd4j/nd4j-backends/nd4j-api-parent/nd4j-api/src/main/java/org/nd4j/linalg/factory/Nd4j.java [hunk 3]
```diff
@@ -992,102 +954,173 @@ public static NDArrayFactory factory() {
...
 
+    /**
+     * The factory used for creating sparse arrays.
+     * @return the factory used for creating sparse arrays.
+     */
     public static NDArrayFactory sparseFactory() {
         return SPARSE_INSTANCE;
     }
 
+    // TODO: We forward the following methods with a default dimension argument. Doc should say something about that.
```
[the remark this answers is not in the document]
Comment (RobAltena) [reply nd4j/nd4j-backends/nd4j-api-parent/nd4j-api/src/main/java/org/nd4j/linalg/factory/Nd4j.java]: Updating just the `cumsum  `docs in the next PR.  Will keep the todo and expand to all methods once it passes review.

File: nd4j/nd4j-backends/nd4j-api-parent/nd4j-api/src/main/java/org/nd4j/linalg/factory/Nd4j.java [hunk 4]
```diff
@@ -587,17 +556,27 @@ public static void setConvolution(ConvolutionInstance convolutionInstance) {
...
         int slices = arrays.length;
         INDArray ret = Nd4j.create(ArrayUtil.combine(new int[] {slices}, sliceShape));
         for (int i = 0; i < ret.slices(); i++)
             ret.putSlice(i, Nd4j.create(arrays[i]).reshape(ArrayUtil.toLongArray(sliceShape)));
         return ret;
     }
 
+    /**
+     * @see #create(LongShapeDescriptor)
+     */
```
[the remark this answers is not in the document]
Comment (RobAltena) [reply nd4j/nd4j-backends/nd4j-api-parent/nd4j-api/src/main/java/org/nd4j/linalg/factory/Nd4j.java]: [PR with fix](https://github.com/SkymindIO/deeplearning4j/pull/76)

File: nd4j/nd4j-backends/nd4j-api-parent/nd4j-api/src/main/java/org/nd4j/linalg/factory/Nd4j.java [hunk 5]
```diff
@@ -677,26 +651,21 @@ public static void setParams(INDArray theta, Collection<INDArray>... matrices) {
...
-
     }
 
-
     /**
-     *
-     * @param arr
-     * @param dimension
-     * @return
+     * Get the maximum (argmax) or minimum (argmin) values for a dimension.
```
[the remark this answers is not in the document]
Comment (RobAltena) [reply nd4j/nd4j-backends/nd4j-api-parent/nd4j-api/src/main/java/org/nd4j/linalg/factory/Nd4j.java]: [PR with fix](https://github.com/SkymindIO/deeplearning4j/pull/76)

File: nd4j/nd4j-backends/nd4j-api-parent/nd4j-api/src/main/java/org/nd4j/linalg/factory/Nd4j.java (hunk 3 above)
[the remark this answers is not in the document]
Comment (RobAltena) [reply nd4j/nd4j-backends/nd4j-api-parent/nd4j-api/src/main/java/org/nd4j/linalg/factory/Nd4j.java]: [PR with fix](https://github.com/SkymindIO/deeplearning4j/pull/76)

File: nd4j/nd4j-backends/nd4j-api-parent/nd4j-api/src/main/java/org/nd4j/linalg/factory/Nd4j.java [hunk 6]
```diff
@@ -1333,7 +1377,13 @@ else if (type == DataType.DOUBLE)
...
     }
 
-
+    /**
+     * Create a buffer equal of length prod(shape)
+     *
+     * @param shape the shape of the buffer to create
+     * @param type the opType to create
+     * @return
+     */
```
[the remark this answers is not in the document]
Comment (RobAltena) [reply nd4j/nd4j-backends/nd4j-api-parent/nd4j-api/src/main/java/org/nd4j/linalg/factory/Nd4j.java]: [PR with fix](https://github.com/SkymindIO/deeplearning4j/pull/76)

**Pull request merged**